src/components/SearchBar.vue

```diff
@@ -0,0 +1,55 @@
+<template>
+    <div class="border-2 border-black mx-3.5 rounded-lg flex py-2 justify-between px-2">
+        <div class="flex mr-2 w-full">
+            <img src="../assets/icons/search.svg" class="pr-2">
+            <input
+                v-model="searchString" 
+                type="text" placeholder="Search todos"
+                class="w-full"
+            >
+        </div>
+        <div class="flex space-x-2 font-semibold items-center">
+            <PrimaryButton
+                v-if="isClearBtnShown"
+                class="w-4 h-4 bg-red rounded-full py-2 content-center text-xs flex items-center"
+                @click="clearTextField"
+            >
+                <p class="pl-1">X</p>
+            </PrimaryButton>
+            <PrimaryButton
+                class="bg-black rounded-xl p-2"
+                @click="search"
```

Comment:
You might want to consider adding a disabled or/and a loading state to this component, just so the user does not press the button to trigger search for example while you are in the middle of an api request, or some time sensitive operation.